modules/apps/knowledge-base/knowledge-base-service/src/main/java/com/liferay/knowledge/base/internal/upgrade/v4_4_0/RSSConfigurationUpgradeProcess.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.knowledge.base.internal.upgrade.v4_4_0;
+
+import com.liferay.knowledge.base.configuration.KBGroupServiceConfiguration;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.GetterUtil;
+
+import java.util.Dictionary;
+
+import org.osgi.framework.Constants;
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.cm.ConfigurationAdmin;
+
+/**
+ * @author Marco Galluzzi
+ */
+public class RSSConfigurationUpgradeProcess extends UpgradeProcess {
+
+	public RSSConfigurationUpgradeProcess(
+		ConfigurationAdmin configurationAdmin) {
+
+		_configurationAdmin = configurationAdmin;
+	}
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		_upgradeRSSConfiguration();
+	}
+
+	private void _upgradeRSSConfiguration() throws Exception {
+		String filterString = String.format(
+			"(%s=%s)", Constants.SERVICE_PID,
+			KBGroupServiceConfiguration.class.getName());
```

Comment:
This only upgrades the system configuration; the ones created for each group won't be taken into consideration. Scoped configurations have different servicePIDs (e.g. "{:systemConfigPID}.scoped~{:uuid}"). Check the OSGi specification for the filter syntax to see how to match these: https://osgi.github.io/osgi/core/framework.module.html#framework.module.filtersyntax.

Reply:
Yes, I've changed the filter to take into consideration multiple configurations (adding an `* ` at the end). Also I'm using the hardcoded name of the `KBGroupServiceConfiguration` class. This was a topic raised during a backend meeting. This upgrade process is meant for a particular service configuration, which is stored in DB. If the class name changes the DB value will not change, and hence it seems more appropriate to have this name hardcoded.